e2e/page-elements/toast.ts

```diff
@@ -24,22 +24,34 @@ export class Toast implements Locatable {
   }
 
   get messages(): Promise<ToastContent[]> {
-    const waitConditions = ExpectedConditions.and(
+    return this.awaitFirstToastPresence()
+      .then(() => toPromise(this.toastLocators
+        .map(async toastElement => new ToastContent(
+          await toPromise(toastElement.getText()),
+          await this.elementHasClass(toastElement, 'toast-success')
+        ))
+      ))
+      .then(promises => Promise.all(promises));
+  }
+
+  public awaitPresence(): Promise<void> {
+    return this.parentElement.awaitPresence().then(() => promisePresenceOf(this.locator));
+  }
+
+  public awaitFirstToastPresence(): Promise<void> {
+    return toPromise(browser.wait(ExpectedConditions.and(
       ExpectedConditions.presenceOf(this.toastLocators.first()),
       ExpectedConditions.visibilityOf(this.toastLocators.first())
-    );
-    return toPromise(browser.wait(waitConditions))
-      .then(() => toPromise(this.toastLocators.map<ToastContent>(async toastElement => {
-        const text = await toastElement.getText();
-        const success = await this.elementHasClass(toastElement, 'toast-success');
-        return new ToastContent(text, success);
-      })));
+    )));
+  }
+
+  public awaitFullPresence(): Promise<void> {
```

Comment:
ehw...i'm confused about full presence, first presence  and presence ... let's thing together of a better naming, ok :wink:  - i think await presence is fine - but full presence should be sth like awaitFullyLoaded